src/main/java/pilcrow/Task.java

```diff
@@ -0,0 +1,232 @@
+        Task task;
+        if (taskType.equals("todo")) {
+            task = new ToDo(taskName, isDone);
+        } else if (taskType.equals("deadline")) {
+            task = new Deadline(taskName, isDone);
+        } else if (taskType.equals("event")) {
+            task = new Event(taskName, isDone);
+        } else {
+            // If not any of the task subtypes
+            // Should never reach this point
+            task = new Task("", false);
+        }
+        return task;
+    }
+
+    /**
+     * Converted a String that has been stored in Storage into a Task.
+     * @param storedTask String representing the stored task in Storage.
+     * @return Converted Task
+     */
+    public static Task convertFromStoredTask(String storedTask) {
+        Task task;
+        Parser parser = new Parser(storedTask);
+        String commandWord = parser.getCommandWord();
+        String restOfCommand = parser.getRestOfCommand().substring(1);
+        Boolean isDone = parser.getRestOfCommand().charAt(0) == 'T';
+
+        task = Task.createTask(commandWord, restOfCommand, isDone);
+        return task;
+    }
+
+    /**
+     * Dummy method, overridden by methods in subclasses.
+     * @return Dummy task
+     */
+    public String convertToStoredTask() {
+        String storedTask = "";
+        return storedTask;
+    }
+
+    public Boolean containsString(String string) {
+        return this.taskName.contains(string);
+    }
+
+    public void setIsDone(Boolean isDone) {
+        this.isDone = isDone;
+    }
+
+    public String getTaskName() {
+        return this.taskName;
+    }
+
+    public Boolean getIsDone() {
+        return this.isDone;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        String taskAsString;
+        taskAsString = "[" + (this.isDone ? 'X' : ' ') + "] " + this.taskName;
+        return taskAsString;
+    }
+
+    /**
+     * Abstraction to represent a Todo Task
+     */
+    private static class ToDo extends Task {
+        private ToDo(String taskName, Boolean isDone) {
+            super(taskName, isDone);
+        }
+
+        /**
+         * Converts a ToDo into String form to be stored.
+         * @return String representing ToDo.
+         */
+        @Override
+        public String convertToStoredTask() {
+            String storedTask = "todo " + (this.isDone ? 'T' : 'F') + this.taskName;
+            return storedTask;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String toString() {
+            String taskAsString;
+            taskAsString = "[T] " + super.toString();
+            return taskAsString;
+        }
+    }
+
+    /**
+     * Abstraction to represent a Deadline Task
+     */
+    private static class Deadline extends Task {
+        private LocalDate deadline;
+
+        private Deadline(String taskName, Boolean isDone) {
```

Comment:
Maybe you should specify the InvalidInputExcetion thrown in the constructor signature? Since it is not being caught and handled inside the constructor itself.

```java
private Deadline(String taskName, Boolean isDone) throws InvalidInputException {
    // ...
}
```